pkg/controller/secretgenerator/secretgenerator_controller.go

```diff
@@ -138,3 +154,52 @@ func (r *ReconcileSecretGenerator) Reconcile(request reconcile.Request) (reconci
 	// Don't requeue if secret is created successfully
 	return reconcile.Result{}, err
 }
+
+func (r *ReconcileSecretGenerator) createOAuthClient(foundSecret *v1.Secret, secret *Secret) error {
+	// Note: Here the route namespace is assumed to be same as the user secret.
+	routeNamespaced := types.NamespacedName{Name: foundSecret.GetAnnotations()[SECRET_OAUTH_CLIENT_ANNOTATION],
+		Namespace: foundSecret.Namespace}
+	oauthClientRoute := &routev1.Route{}
+	// Get spec.host from route
+	err := wait.PollImmediate(resourceRetryInterval, resourceRetryTimeout, func() (done bool, err error) {
+		err = r.client.Get(context.TODO(), routeNamespaced, oauthClientRoute)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		} else {
+			return true, nil
```

Comment:
If the route exists we should check if the `.spec.host` exists and return false if not to retry getting it again. The host is added by the Openshift ingress controller, there is a corner case in which the route could exist but the host doesn't yet.